### ISSUE TITLE
D3D compile failed

### ISSUE THREAD

**Issue by cx20** [Twinklebear/webgpu-volume-raycaster#1]
I tried the following sample on Chrome Canary.
https://www.willusher.io/webgpu-volume-raycaster/

However, I could not display the sample with the following error. Is there a workaround?
```
D3D compile failed with C:\Users\xxxx\AppData\Local\Google\Chrome SxS\Application\95.0.4624.4\Shader@0x00003A6E0030F800(96,30-82): warning X3570: gradient instruction used in a loop with varying iteration, attempting to unroll the loop
C:\Users\xxxx\AppData\Local\Google\Chrome SxS\Application\95.0.4624.4\Shader@0x00003A6E0030F800(95,5-137): error X3511: unable to unroll loop, loop does not appear to terminate in a timely manner (857 iterations) or unrolled loop is too large, use the [unroll(n)] attribute to force an exact higher number

    at CompileShaderFXC (../../third_party/dawn/src/dawn_native/d3d12/ShaderModuleD3D12.cpp:159)
    at operator() (../../third_party/dawn/src/dawn_native/d3d12/ShaderModuleD3D12.cpp:356)
    at GetOrCreate (../../third_party/dawn/src\dawn_native/PersistentCache.h:73)
    at Compile (../../third_party/dawn/src/dawn_native/d3d12/ShaderModuleD3D12.cpp:362)
    at Initialize (../../third_party/dawn/src/dawn_native/d3d12/RenderPipelineD3D12.cpp:369)
    at Create (../../third_party/dawn/src/dawn_native/d3d12/RenderPipelineD3D12.cpp:331)
    at GetOrCreateRenderPipeline (../../third_party/dawn/src/dawn_native/Device.cpp:666)
250Object is an error.
    at ValidateObject (..<URL>)
    at operator() (..<URL>)
    at FinishInternal (..<URL>)
249Object is an error.
    at ValidateObject (..<URL>)
    at ValidateSubmit (..<URL>)
www.willusher.io/:1 WebGPU: too many warnings, no more warnings will be reported to the console for this GPUDevice.
```
Environment : WIndows 10 + GeForce RTX 2060 + Chrome Canary 95.

**Comment by Twinklebear** [comment]
Ah shoot thanks for opening this issue, I only ran this on my Mac so didn't see this but the loop needs to be rewritten a bit so that DX can see it will terminate. I'll fix that up in the next few days

**Comment by Twinklebear** [comment]
This should be fixed in af333f6e7a14cf44804fd0e621f1c51f6baedb67 , I just updated the online demo as well which can now run on my windows laptop. Let me know how it looks!
